apps/prairielearn/src/lib/score-upload.ts

```diff
@@ -394,28 +394,37 @@ async function updateAssessmentInstanceFromJson(
   assessment_id: string,
   authn_user_id: string,
 ) {
-  let query, id;
-  if (_.has(json, 'uid')) {
-    query = sql.select_assessment_instance_uid;
-    id = json.uid;
-  } else if (_.has(json, 'group_name')) {
-    query = sql.select_assessment_instance_group;
-    id = json.group_name;
-  } else {
-    throw new Error('"uid" or "group_name" not found');
-  }
   if (!_.has(json, 'instance')) throw new Error('"instance" not found');
-
   await sqldb.runInTransactionAsync(async () => {
-    const assessment_instance_id = await sqldb.queryOptionalRow(
-      query,
-      {
-        assessment_id,
-        uid: json.uid,
-        instance_number: json.instance,
-      },
-      IdSchema,
-    );
+    let assessment_instance_id: string | null;
+    let id: string | null;
+    if (_.has(json, 'uid')) {
+      id = json.uid;
+      assessment_instance_id = await sqldb.queryOptionalRow(
+        sql.select_assessment_instance_uid,
+        {
+          assessment_id,
+          uid: json.uid,
+          instance_number: json.instance,
+        },
+        IdSchema,
+      );
+    } else if (_.has(json, 'group_name')) {
+      id = json.group_name;
+      assessment_instance_id = await sqldb.queryOptionalRow(
+        sql.select_assessment_instance_group,
+        {
+          assessment_id,
+          uid: json.uid,
```

Comment:
I believe this is not used, right?

Reply:
Right, I'll fix that too. I guess I was expecting a test to fail but maybe we never run something on that branch.